usr/share/rear/lib/uefi-functions.sh

```diff
@@ -39,46 +39,73 @@ function trim {
+            else
+                Error "Can not determine partition modules, ${dirs[*]} would be likely inaccessible in GRUB2"
+            fi
+        fi
+
+        if [ -n "$gprobe" ]; then
+            # this is unfortunately only a crude approximation of the Grub internal probe_mods() function
+            modules=( $( for p in "${dirs[@]}" ; do
+                             $gprobe --target=fs "$p"
+                             $gprobe --target=partmap "$p" | sed -e 's/^/part_/'
+                             $gprobe --target=abstraction "$p"
+                         done | sort -u ) )
+        fi
+        Log "GRUB2 modules to load: ${modules:+${modules[*]}}"
     fi
+
     # grub-mkimage needs /usr/lib/grub/x86_64-efi/moddep.lst (cf. https://github.com/rear/rear/issues/1193)
     # and at least on SUSE systems grub2-mkimage needs /usr/lib/grub2/x86_64-efi/moddep.lst (in 'grub2' directory)
     # so that we error out if grub-mkimage or grub2-mkimage would fail when its moddep.lst is missing.
     # Careful: usr/sbin/rear sets nullglob so that /usr/lib/grub*/x86_64-efi/moddep.lst gets empty if nothing matches
     # and 'test -f' succeeds with empty argument so that we cannot use 'test -f /usr/lib/grub*/x86_64-efi/moddep.lst'
     # also 'test -n' succeeds with empty argument but (fortunately/intentionally?) plain 'test' fails with empty argument:
-    test /usr/lib/grub*/x86_64-efi/moddep.lst || Error "$gmkimage would not make bootable EFI image of GRUB2 (no /usr/lib/grub*/x86_64-efi/moddep.lst file)"
-    # As not all Linux distros have the same GRUB2 modules present we verify what we have (see also https://github.com/rear/rear/pull/2001)
-    local grub_module=""
-    local grub_modules=""
-
-    # Use a reasonable fallback value, cf. https://github.com/rear/rear/pull/2283
-    test "$GRUB2_MODULES" || GRUB2_MODULES="part_gpt part_msdos fat ext2 \
-normal chain boot configfile linux linuxefi multiboot jfs iso9660 usb \
-usbms usb_keyboard video udf ntfs all_video gzio efi_gop reboot search \
-test echo btrfs lvm"
-
-    LogPrint "Using '$GRUB2_MODULES' modules to build $outfile"
-
-    for grub_module in $GRUB2_MODULES ; do
-        test "$( find /usr/lib/grub*/x86_64-efi -type f -name "$grub_module.mod" 2>/dev/null )" && grub_modules="$grub_modules $grub_module"
-    done
-    if ! $gmkimage $v -O x86_64-efi $embedded_config -o $outfile -p "/EFI/BOOT" $grub_modules ; then
-        Error "Failed to make bootable EFI image of GRUB2 (error during $gmkimage of $outfile)"
+    test /usr/lib/grub*/x86_64-efi/moddep.lst || Error "$gmkstandalone would not make bootable EFI image of GRUB2 (no /usr/lib/grub*/x86_64-efi/moddep.lst file)"
+    if ! $gmkstandalone $v ${GRUB2_MODULES:+"--install-modules=${GRUB2_MODULES[*]}"} ${modules:+"--modules=${modules[*]}"} -O x86_64-efi -o $outfile $embedded_config ; then
+        Error "Failed to make bootable EFI image of GRUB2 (error during $gmkstandalone of BOOTX64.efi)"
```

Comment:
`BOOTX64.efi` should be replaced by `$outfile` here

Reply:
Corrected in the rebased version